docs/componentExtension.md

```diff
@@ -0,0 +1,83 @@
+## Customizing and Extending Components
+
+As mentioned in the [architecture document](architecture.md), when customizing or extending the sample app component, do not modify the packages within `@sfcc-bff` and `@sfcc-core`. These packages will be published and consumed via `npm`. Instead, create a new custom package within the monorepo that registers itself with `@sfcc-core\core` and provides access to data from a third-party service.
```

Comment:
We might be less authoritative, and say 'you should not modify...' instead of 'do not modify...'. Moreover, the new package should be either part of the application or a partner can create his own module with the extension.